src/status.js

```diff
@@ -254,62 +263,268 @@
-							service: 'status',
-							func: 'StatusChanged',
-							obj: {
-								interface: 'fr.partnering.Status.Part',
-								path: objectPath
-							},
-							data: robotNames
-						}, (peerId, err, data) => {
-							if (err != null) {
-								Logger.error("StatusSubscribe:" + err);
-							} else {
-								sendData[0] = data;
-								this._getRobotModelFromRecv2(sendData, robotId, robotName);
-								if (typeof callback === 'function') {
-									callback(this.robotModel, peerId);
-								}
-							}
-						});
-						this.subscriptions.push(subs);
-					}
+				if (err != null || objData == null) {
+					reject(err)
 				}
+				resolve(objData) // returns a map that links the object path to its corresponding interface
 			})
-		}).catch(err => {
-			Logger.error(err);
 		})
 
+		let robotIface = 'fr.partnering.Status.Robot'
+		let partIface = 'fr.partnering.Status.Part'
+
+		// js objects of robots and parts
+		let robotMap = new Map() // map robot name to id
+		let robots = [] // list of robot objects
+		let parts = [] // list of part object
+		let meshedRobotNames = [] // list of names of robots and diya-server in the mesh network
+
+		// Retrieve reference map of keys and values in order to reduce payload for status requests
+		return Promise.try(_ => this._getPartReferenceMap())
+			.then(_ => this._getStatusEvtReferenceMap())
+			.then(_ => getNeighbors)
+			.then(ret => {
+				if (ret == null || !Array.isArray(ret)) {
+					meshedRobotNames = []
+				}
+				let hostname = this.selector._connection._self
+				meshedRobotNames = ret.map(r => r[0]) // we only keep the robot names
+				if (!meshedRobotNames.includes(hostname)) {
+					meshedRobotNames.push(hostname) // add hostname, i.e. the diya-server, which is not in the list of neighbors
+				}
+			})
+			.then(_ => getRobotsAndParts)
+			.then(ret => {
+				for (let objectPath in ret) {
+					// the object obtained from the object path
+					let object = ret[objectPath]
+
+					// if the return object is of a robot in the list of neighbors, or of the diya-server, retrieve all ofits relevant statuses
+					if (object.hasOwnProperty(robotIface)) { // this is robot object
+						robots.push(object[robotIface])
+					}
+
+					// if the return object is of a part, listen to signal StatusChanged of the part
+					if (object.hasOwnProperty(partIface)) { // this is a part object
+						let part = object[partIface]
+						part.objectPath = objectPath
+						part.RobotName = objectPath.split('/')[5] /* /fr/partnering/Status/Robots/B1R00037/Parts/voct */
+						parts.push(part)
+					}
+				}
+
+				Logger.debug('robots', robots)
+				Logger.debug('parts', parts)
+
+				// filer and keep the diya-server and the robots that are only in the same mesh networks
+				robots = robots.filter(robot => meshedRobotNames.includes(robot.RobotName)) // only keeps robots that are neighbors (i.e. in the same mesh network)
```

Comment:
I do not understand how you could see a robot that is not in the same mesh network. Do you mean connected ?